internal/storage/v2/clickhouse/wrapper/wrapper.go

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2025 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package wrapper
+
+import (
+	"context"
+
+	"github.com/ClickHouse/ch-go"
+	"github.com/ClickHouse/ch-go/chpool"
+	v2Client "github.com/ClickHouse/clickhouse-go/v2"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/internal/storage/v2/clickhouse/client"
+	"github.com/jaegertracing/jaeger/internal/storage/v2/clickhouse/config"
+	"github.com/jaegertracing/jaeger/internal/storage/v2/clickhouse/tracestore/dbmodel"
+)
+
+// PoolWrapper is a wrapper around ch-go used by write path.
+type PoolWrapper struct {
```

Comment:
I am not convinced the wrapper and client abstractions are really necessary. There are always other ways to mock them for tests, for example:
```go
type Writer struct {
  execQuery func(ctx context.Context, query string, ch.Input) error
}

func NewWriter(pool *chpool.Pool) *Writer {
  return &Writer{
    execQuery: pool.Do,
  }
}
```

Here you can still mock execQuery function in the writer's tests to validate the business logic.